hack/generator/pkg/astmodel/storage_conversion_function.go

```diff
@@ -0,0 +1,414 @@
+)
+
+// Ensure that StorageConversionFunction implements Function
+var _ Function = &StorageConversionFunction{}
+
+// NewStorageConversionFromFunction creates a new StorageConversionFunction to convert from the specified source
+func NewStorageConversionFromFunction(
+	receiver TypeDefinition,
+	sourceHubType TypeDefinition,
+	intermediateType *TypeDefinition,
+	idFactory IdentifierFactory,
+) (*StorageConversionFunction, error) {
+	result := &StorageConversionFunction{
+		name:                "ConvertFrom",
+		hubType:             sourceHubType,
+		intermediateType:    intermediateType,
+		idFactory:           idFactory,
+		conversionDirection: ConvertFrom,
+		conversions:         make(map[string]StoragePropertyConversion),
+		knownLocals:         NewKnownLocalsSet(idFactory),
+	}
+
+	err := result.createConversions(receiver)
+	return result, err
+}
+
+// NewStorageConversionToFunction creates a new StorageConversionFunction to convert to the specified destination
+func NewStorageConversionToFunction(
+	receiver TypeDefinition,
+	destinationHubType TypeDefinition,
+	intermediateType *TypeDefinition,
+	idFactory IdentifierFactory,
+) (*StorageConversionFunction, error) {
+	result := &StorageConversionFunction{
+		name:                "ConvertTo",
+		hubType:             destinationHubType,
+		intermediateType:    intermediateType,
+		idFactory:           idFactory,
+		conversionDirection: ConvertTo,
+		conversions:         make(map[string]StoragePropertyConversion),
+		knownLocals:         NewKnownLocalsSet(idFactory),
+	}
+
+	err := result.createConversions(receiver)
+	return result, err
+}
+
+// Name returns the name of this function
+func (fn *StorageConversionFunction) Name() string {
+	return fn.name
+}
+
+// RequiredPackageReferences returns the set of package references required by this function
+func (fn *StorageConversionFunction) RequiredPackageReferences() *PackageReferenceSet {
+	result := NewPackageReferenceSet(fn.hubType.Name().PackageReference)
+
+	if fn.intermediateType != nil {
+		result.AddReference(fn.intermediateType.Name().PackageReference)
+	}
+
+	return result
+}
+
+// References returns the set of types referenced by this function
+func (fn *StorageConversionFunction) References() TypeNameSet {
+	result := NewTypeNameSet(fn.hubType.Name())
+
+	if fn.intermediateType != nil {
+		result.Add(fn.intermediateType.Name())
+	}
+
+	return result
+}
+
+// Equals checks to see if the supplied function is the same as this one
+func (fn *StorageConversionFunction) Equals(f Function) bool {
+	if other, ok := f.(*StorageConversionFunction); ok {
+		if fn.name != other.name {
+			// Different name means not-equal
+			return false
+		}
+
+		if len(fn.conversions) != len(other.conversions) {
+			// Different count of conversions means not-equal
+			return false
+		}
+
+		for name := range fn.conversions {
+			if _, found := other.conversions[name]; !found {
+				// Missing conversion means not-equal
+				return false
+			}
+		}
```

Comment:
I think you're missing a `return true` here - without it this function never returns true